src/pages/GroupChatNameEditPage.tsx

```diff
@@ -0,0 +1,119 @@
+
+type GroupChatNameEditPageOnyxProps = {
+    groupChatDraft: NewGroupChatDraft | null;
+};
+
+type GroupChatNameEditPageProps = StackScreenProps<NewChatNavigatorParamList, typeof SCREENS.NEW_CHAT.NEW_CHAT_EDIT_NAME> & GroupChatNameEditPageOnyxProps;
+
+function GroupChatNameEditPage({groupChatDraft, route}: GroupChatNameEditPageProps) {
+    // If we have a reportID this means we are using this page to update an existing Group Chat name
+    const reportID = route.params?.reportID ?? '';
+    const isUpdatingExistingReport = Boolean(reportID);
+
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const {inputCallbackRef} = useAutoFocusInput();
+
+    // We will try to get the chatName from the report or draft depending on what flow we are in
+    const participantAccountIDs = useMemo(() => {
+        if (reportID) {
+            return ReportUtils.getParticipantAccountIDs(reportID);
+        }
+
+        return (groupChatDraft?.participants ?? []).map((participant) => participant.accountID);
+    }, [groupChatDraft, reportID]);
+    const existingReportName = useMemo(() => ReportUtils.getGroupChatName(participantAccountIDs, false, reportID), [participantAccountIDs, reportID]);
+    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+    const currentChatName = reportID ? existingReportName : groupChatDraft?.reportName || existingReportName;
+
+    const validate = useCallback(
+        (values: FormOnyxValues<typeof ONYXKEYS.FORMS.NEW_CHAT_NAME_FORM>): Errors => {
+            const errors: Errors = {};
+            if (!ValidationUtils.isValidReportName(values[INPUT_IDS.NEW_CHAT_NAME] ?? '')) {
+                errors.newChatName = translate('common.error.characterLimit', {limit: CONST.REPORT_NAME_LIMIT});
+            }
+
+            return errors;
+        },
+        [translate],
+    );
+
+    const editName = useCallback(
+        (values: FormOnyxValues<typeof ONYXKEYS.FORMS.NEW_CHAT_NAME_FORM>) => {
+            if (isUpdatingExistingReport) {
+                Report.updateGroupChatName(reportID, values[INPUT_IDS.NEW_CHAT_NAME] ?? '');
+                Navigation.goBack(ROUTES.REPORT_SETTINGS.getRoute(reportID));
+                return;
+            }
+
+            Report.setGroupDraft({reportName: values[INPUT_IDS.NEW_CHAT_NAME]});
+            Navigation.goBack(ROUTES.NEW_CHAT_CONFIRM);
+        },
+        [isUpdatingExistingReport, reportID],
+    );
+
+    return (
+        <ScreenWrapper
+            includeSafeAreaPaddingBottom={false}
+            style={[styles.defaultModalContainer]}
+            testID={GroupChatNameEditPage.displayName}
+            shouldEnableMaxHeight
+        >
+            <HeaderWithBackButton
+                title={translate('groupConfirmPage.groupName')}
+                onBackButtonPress={() => Navigation.goBack(isUpdatingExistingReport ? ROUTES.REPORT_SETTINGS.getRoute(reportID) : ROUTES.NEW_CHAT_CONFIRM)}
```

Comment:
Hey folks👋
This has caused https://github.com/Expensify/App/issues/45812, essentially, you vere navigated back to the wrong page after refreshing GroupChatNameEditPage. This was resolved by using `ROUTES.REPORT_WITH_ID_DETAILS` instead of `ROUTES.REPORT_SETTINGS`